### PR TITLE
Switch to pathlib

### PR DESCRIPTION
**Pull request by eric-nieuwland**
Use pathlib instead of os.path and open().
Some additional testing required as I don't have all the tools.

### CODE REVIEW

**Comment by fniessink** [comment]
Can you fix the automated review comments here: https://github.com/ICTU/quality-report/pull/426? The tests are all run by Travis. See: https://travis-ci.org/ICTU/quality-report/builds/391265268.

File: docs/examples/quality_report/project_definition.py [hunk 1]
```diff
@@ -9,17 +9,20 @@
 hqlib.metric_source.url_opener.UrlOpener = UrlOpenerMock
 ### END
 
+import pathlib
```
[the remark this answers is not in the document]
Comment (eric-nieuwland) [reply docs/examples/quality_report/project_definition.py]: WONT FIX matches original

File: backend/hqlib/filesystem.py [hunk 2]
```diff
@@ -25,24 +23,48 @@ class FileSystem(object):
...
+        if not dir_path.exists():
+            dir_path.mkdir(parents=True)
+        dir_path.chmod(stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH | stat.S_IRUSR | stat.S_IWUSR)
 
+    @typing.overload
     @staticmethod
     def make_file_readable(filename: str) -> None:
+        pass
+
+    @typing.overload
```
[the remark this answers is not in the document]
Comment (eric-nieuwland) [reply backend/hqlib/filesystem.py]: WONT FIX as advised by documentation of the 'typing' module

File: backend/hqlib/filesystem.py [hunk 3]
```diff
@@ -25,24 +23,48 @@ class FileSystem(object):
...
+    @staticmethod
+    def make_file_readable(filename: pathlib.Path) -> None:
+        pass
+
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (eric-nieuwland) [reply backend/hqlib/filesystem.py]: WONT FIX as advised by documentation of the 'typing' module

File: backend/hqlib/filesystem.py [hunk 4]
```diff
@@ -25,24 +23,48 @@ class FileSystem(object):
...
     @staticmethod
     def create_dir(dir_name: str) -> None:
+        pass
+
+    @typing.overload
+    @staticmethod
+    def create_dir(dir_name: pathlib.Path) -> None:
+        pass
+
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (eric-nieuwland) [reply backend/hqlib/filesystem.py]: WONT FIX as advised by documentation of the 'typing' module

File: backend/hqlib/filesystem.py [hunk 5]
```diff
@@ -25,24 +23,48 @@ class FileSystem(object):
...
+        else:
+            path.write_text(contents, encoding=encoding)
+        FileSystem.make_file_readable(path)
 
+    @typing.overload
     @staticmethod
     def create_dir(dir_name: str) -> None:
+        pass
+
+    @typing.overload
```
[the remark this answers is not in the document]
Comment (eric-nieuwland) [reply backend/hqlib/filesystem.py]: WONT FIX as advised by documentation of the 'typing' module

**Pull request closed without merging**